tools/astarte_e2e/lib/astarte_e2e/client.ex

```diff
@@ -0,0 +1,333 @@
+#
+# This file is part of Astarte.
+#
+# Cospyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteE2E.Client do
+  alias Phoenix.Channels.GenSocketClient
+  alias Phoenix.Channels.GenSocketClient.Transport.WebSocketClient
+
+  require Logger
+
+  @doc "Starts the client process."
+  @spec start_link(AstarteE2E.client_options()) :: GenSocketClient.on_start()
+  def start_link(opts) do
+    transport = WebSocketClient
+    url = Keyword.fetch!(opts, :url)
+    token = Keyword.fetch!(opts, :token)
+    realm = Keyword.fetch!(opts, :realm)
+    device_id = Keyword.fetch!(opts, :device_id)
+
+    verify_option =
+      if Keyword.get(opts, :ignore_ssl_errors, false) do
+        :verify_none
+      else
+        :verify_peer
+      end
+
+    with {:ok, pid} <-
+           GenSocketClient.start_link(
+             __MODULE__,
+             transport,
+             {url, realm, token, device_id},
+             [transport_opts: [ssl_verify: verify_option]],
+             name: :astarte_ws_client
+           ) do
+      Logger.info("[Client] Started process with pid #{inspect(pid)}.")
```

Comment:
Let's avoid writing `[Client]`, let's use pretty log and logfmt instead. Also some logger metadata might be useful.